Exiled.Events/Patches/Events/Player/InsertingGeneratorTablet.cs

```diff
@@ -34,6 +34,38 @@ private static bool Prefix(Generator079 __instance, GameObject person, PlayerInt
                 {
                     case PlayerInteract.Generator079Operations.Door:
                         bool isAllowed = true;
+                        if (!__instance.isDoorUnlocked)
+                        {
+                            isAllowed = person.GetComponent<ServerRoles>().BypassMode;
+                            
+                            if (person.GetComponent<Inventory>().curItem > 0)
+                            {
+                                foreach (string permission in person.GetComponent<Inventory>().GetItemByID(person.GetComponent<Inventory>().curItem).permissions)
+                                {
+                                    if (permission == "ARMORY_LVL_2")
+                                        isAllowed = true;
+                                }
+                            }
+
+                            var unlockingEventArgs = new UnlockingGeneratorEventArgs(API.Features.Player.Get(person), __instance, isAllowed);
```

Comment:
You never check to see if the generator is locked, therefore you will call this event everytime a closed generator door is open, regardless of if it's actually being unlocked.